com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/VFX/Config.template.hlsl

```diff
@@ -0,0 +1,302 @@
+$splice(VFXDefineSpace)
+
+$splice(VFXDefines)
+
+// Explicitly defined here for now (similar to how it was done in the previous VFX code-gen)
+#define HAS_ATTRIBUTES 1
+
+#define VFX_NEEDS_COLOR_INTERPOLATOR (VFX_USE_COLOR_CURRENT || VFX_USE_ALPHA_CURRENT)
+#if HAS_STRIPS
+#define VFX_OPTIONAL_INTERPOLATION
+#else//
+#define VFX_OPTIONAL_INTERPOLATION nointerpolation
+#endif
+
+ByteAddressBuffer attributeBuffer;
+
+#if VFX_HAS_INDIRECT_DRAW
+StructuredBuffer<uint> indirectBuffer;
+#endif
+
+#if USE_DEAD_LIST_COUNT
+ByteAddressBuffer deadListCount;
+#endif
+
+#if HAS_STRIPS
+Buffer<uint> stripDataBuffer;
+#endif
+
+// #if WRITE_MOTION_VECTOR_IN_FORWARD || USE_MOTION_VECTORS_PASS
+ByteAddressBuffer elementToVFXBufferPrevious;
+// #endif
+
+CBUFFER_START(outputParams)
+    float nbMax;
+    float systemSeed;
+CBUFFER_END
+
+// Helper macros to always use a valid instanceID
+#if defined(UNITY_STEREO_INSTANCING_ENABLED)
+    #define VFX_DECLARE_INSTANCE_ID     UNITY_VERTEX_INPUT_INSTANCE_ID
+    #define VFX_GET_INSTANCE_ID(i)      unity_InstanceID
+#else
+    #define VFX_DECLARE_INSTANCE_ID     uint instanceID : SV_InstanceID;
+    #define VFX_GET_INSTANCE_ID(i)      input.instanceID
+#endif
+
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/VFXGraph/Shaders/VFXCommon.hlsl"
+#include "Packages/com.unity.visualeffectgraph/Shaders/VFXCommon.hlsl"
+
+$splice(VFXParameterBuffer)
+
+$splice(VFXGeneratedBlockFunction)
+
+struct AttributesElement
+{
+    uint index;
+    Attributes attributes;
+#if HAS_STRIPS
+    uint relativeIndexInStrip;
+    StripData stripData;
+#endif
+};
+
+bool ShouldCull(uint index)
```

Comment:
side note: the function name: GetSize / ShouldCull are extremely generic and could conflict with users function or thing like that. Maybe worse being carefult with name